### PR TITLE
WIP: Introduce new in-place compaction GC

### PR DESCRIPTION
**Pull request by WenbinZhu**
## Overview

Description:

Why should this be merged: 

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [ ] There are no TODOs left in the code
- [ ] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [ ] Change is covered by automated tests
- [ ] Public API has Javadoc

### CODE REVIEW

File: infrastructure/src/main/java/org/corfudb/infrastructure/log/AbstractLogSegment.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    /**
+     * Opens the file channel to manipulate the file, creating a new file if
+     * it does not exist. If the readOnly parameter is set, the channel will
+     * be opened in read-only mode.
+     *
+     * @param filePath the path of the file to open
+     * @param readOnly if the file should be opened only for read
+     * @return the file channel to manipulate the specified file
+     */
+    private FileChannel getChannel(String filePath, boolean readOnly) throws IOException {
```
Comment (lixinchengdu) [inline infrastructure/src/main/java/org/corfudb/infrastructure/log/AbstractLogSegment.java]: static

File: infrastructure/src/main/java/org/corfudb/infrastructure/log/GarbageLogSegment.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    public GarbageLogSegment(long startAddress, String fileName,
+                             StreamLogParams logParams,
+                             ResourceQuota logSizeQuota,
+                             SegmentMetaData segmentMetaData) {
+        super(startAddress, fileName, logParams, logSizeQuota, segmentMetaData);
+    }
+
+    public void append(long address, LogData entry) {
+
```
Comment (lixinchengdu) [inline infrastructure/src/main/java/org/corfudb/infrastructure/log/GarbageLogSegment.java]: TODO

File: infrastructure/src/main/java/org/corfudb/infrastructure/log/GarbageLogSegment.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    @Override
+    public void close() {
+
```
Comment (lixinchengdu) [inline infrastructure/src/main/java/org/corfudb/infrastructure/log/GarbageLogSegment.java]: TODO

**Comment by WenbinZhu** [comment]
Closing as the injection framework had some issues before, will create a new one.

**Pull request closed without merging**